transform_dialect/examples/accel/mmt4d_codegen_spec.mlir

```diff
@@ -0,0 +1,117 @@
+// CHECK-IR:   memref.subview {{.*}} : memref<16x32x8x8xf32, strided<[4096, 64, 8, 1], offset: ?>> to memref<8x8x8x8xf32, strided<[4096, 64, 8, 1], offset: ?>>
+// CHECK-IR:   linalg.generic {indexing_maps = [#map2, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins{{.*}} outs(%alloca : memref<8x8x8x8xf32>) { linalg.yield }
+// CHECK-IR:   linalg.mmt4d ins({{.*}}, {{.*}} : memref<8x8x8x1xf32, strided<[64, 8, 1, 1], offset: ?>>, memref<8x8x8x1xf32, strided<[64, 8, 1, 1], offset: ?>>) outs(%alloca : memref<8x8x8x8xf32>)
+// CHECK-IR:   linalg.generic {indexing_maps = [#map2, #map2], iterator_types = ["parallel", "parallel", "parallel", "parallel"]} ins(%alloca : memref<8x8x8x8xf32>) outs({{.*}} : memref<8x8x8x8xf32, strided<[4096, 64, 8, 1], offset: ?>>) { linalg.yield }
+// CHECK-IR:   gpu.barrier
+// CHECK-IR:  }
+// CHECK-IR:  return
+// CHECK-IR: }
+
+module attributes { transform.with_named_sequence } {
+  transform.named_sequence @cleanup(%variant_op: !transform.any_op {transform.readonly}) {
+    %func = transform.structured.match ops{["func.func"]} in %variant_op : (!transform.any_op) -> !transform.any_op
+      transform.apply_patterns to %func {
+      transform.apply_patterns.linalg.tiling_canonicalization
+      transform.apply_patterns.iree.fold_fill_into_pad
+      transform.apply_patterns.scf.for_loop_canonicalization
+      transform.apply_patterns.canonicalization
+    } : !transform.any_op
+    transform.iree.apply_licm %func : !transform.any_op
+    transform.iree.apply_cse %func : !transform.any_op
+    transform.yield
+  }
+
+  transform.sequence failures(propagate) {
+  ^bb1(%variant_op: !transform.any_op):
+    %matmul = transform.structured.match ops{["linalg.mmt4d"]} in %variant_op : (!transform.any_op) -> !transform.any_op
+
+    // First level tile to forall with tile_sizes [16, 32].
+    %forall, %tiled_matmul =
+      transform.structured.tile_to_forall_op %matmul tile_sizes [16, 32]
+        ( mapping = [#gpu.block<y>, #gpu.block<x>] ) : (!transform.any_op) -> (!transform.any_op, !transform.any_op)
+    transform.iree.populate_workgroup_count_region_using_num_threads_slice %forall
+      : (!transform.any_op) -> ()
+
+    // Tile reduction dimension.
+    %tiled_reduction, %loop =
+      transform.structured.tile %tiled_matmul [0, 0, 8]
+      : (!transform.any_op) -> (!transform.any_op, !transform.any_op)
+
+    // Pad operation.
+    %padded, %pad, %__ = transform.structured.pad %tiled_reduction {
+      padding_values=[0.0 : f32, 0.0 : f32, 0.0 : f32],
+      padding_dimensions=[0, 1, 2],
+      pack_paddings=[1, 1, 0],
+      copy_back_op="none"
+    } : (!transform.any_op) -> (!transform.any_op, !transform.any_op, !transform.any_op)
+
+    // Second level tile to forall with tile_sizes [8, 8].
+    %forall_1, %tiled_matmul_1 =
+      transform.structured.tile_to_forall_op %padded tile_sizes [8, 8]
```

Comment:
How about we just make this `[1, 1]`. so each shared -> local is to do 1 inner tile?